graphql/src/main/java/io/strimzi/admin/graphql/GraphQLService.java

```diff
@@ -0,0 +1,120 @@
+import io.strimzi.admin.graphql.registration.RuntimeWiringRegistry;
+import io.strimzi.admin.http.server.registration.RouteRegistration;
+import io.strimzi.admin.http.server.registration.RouteRegistrationDescriptor;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.graphql.GraphQLHandler;
+import io.vertx.ext.web.handler.graphql.GraphiQLHandler;
+import io.vertx.ext.web.handler.graphql.GraphiQLHandlerOptions;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class GraphQLService implements RouteRegistration {
+
+    private static final Logger LOGGER = LogManager.getLogger(GraphQLService.class);
+    private static final String BASE_SCHEMA_LOCATION = "graphql-schema/baseSchema.graphql";
+
+    @Override
+    public Future<RouteRegistrationDescriptor> getRegistrationDescriptor(Vertx vertx) {
+        final Promise<RouteRegistrationDescriptor> promise = Promise.promise();
+
+        final Router router = Router.router(vertx);
+        final RouteRegistrationDescriptor graphQLRegistrationDescriptor = RouteRegistrationDescriptor.create("/", router);
+
+        if (GraphiQLHandlerOptions.DEFAULT_ENABLED) {
+            LOGGER.warn("GraphiQL is enabled");
+            router.route("/graphiql/*").handler(GraphiQLHandler.create());
+        }
+
+        configureGraphQLHandler(vertx)
+            .onSuccess(graphQLHandler -> {
+                graphQLHandler.queryContext(routingContext -> routingContext);
+                router.post("/graphql").handler(graphQLHandler);
+                promise.complete(graphQLRegistrationDescriptor);
+            })
+            .onFailure(throwable -> {
+                LOGGER.error("GraphQL service failed to initialize - {} ", throwable.getMessage());
+                promise.fail(throwable);
+            });
+
+        return promise.future();
+    }
+
+    private Future<GraphQLHandler> configureGraphQLHandler(final Vertx vertx) {
+        Promise<GraphQLHandler> promise = Promise.promise();
+
+        vertx.executeBlocking(p -> {
+                final SchemaParser schemaParser = new SchemaParser();
+                final InputStreamReader userInputStream = new InputStreamReader(
+                    Objects.requireNonNull(
+                        getClass()
+                            .getClassLoader()
+                            .getResourceAsStream(GraphQLService.BASE_SCHEMA_LOCATION)),
+                    StandardCharsets.UTF_8);
+
+                final TypeDefinitionRegistry typeDefinitionRegistry = schemaParser.parse(userInputStream);
+                p.complete(typeDefinitionRegistry);
```

Comment:
Might want to wrap this in try and use a catch to fail `p`, since I'm guessing parsing might fail for all the usual reasons.

Reply:
Wrapped this instance and all others using a similar mechanism. Currently catches `Exception` but I was wondering if it should be scoped to i/o exceptions.